spring-integration-core/src/main/java/org/springframework/integration/transformer/SyslogToMapTransformer.java

```diff
@@ -88,6 +90,10 @@ else if (isString) {
 	private Map<String, ?> transform(String payload) {
 		Map<String, Object> map = new LinkedHashMap<String, Object>();
 		Matcher matcher = pattern.matcher(payload);
+		SimpleDateFormat dateFormat = dateFormats.poll();
+		if (dateFormat == null) {
+			dateFormat = new SimpleDateFormat("MMM dd HH:mm:ss");
```

Comment:
That is an interesting technique. I wonder if we can optimize it a bit with several pre-populated instances from the ctor.
From other side: shouldn't we consider an `OutOfMemoryError` in case of high-load?
E.g. do the stuff with `SoftReference<SimpleDateFormat>`?

Or... `RingBuffer` to have control over memory and avoid any GC?

No need to do anything from my comments. Just in case to think a bit...